src/ctap/pin_protocol_v1.rs

```diff
@@ -136,12 +136,31 @@ fn check_and_store_new_pin(
         // TODO(kaczmarczyck) check last byte == 0x00
         return false;
     }
-    let mut pin_hash = [0; 16];
+    let mut pin_hash = [0u8; 16];
     pin_hash.copy_from_slice(&Sha256::hash(&pin[..])[..16]);
     persistent_store.set_pin_hash(&pin_hash);
     true
 }
 
+#[cfg(feature = "with_ctap2_1")]
+// TODO remove when all variants are used
+#[allow(dead_code)]
+pub enum PinPermission {
+    MakeCredential = 0x01,
+    GetAssertion = 0x02,
+    CredentialManagement = 0x04,
+    BioEnrollment = 0x08,
+    PlatformConfiguration = 0x10,
+    AuthenticatorConfiguration = 0x20,
+}
+
+#[cfg(feature = "with_ctap2_1")]
+impl PinPermission {
+    pub fn check(self, stored_bits: u8) -> bool {
+        self as u8 & stored_bits != 0
```

Comment:
Just making sure. This relies on the fact that PinPermission is exactly a bitflag (i.e. each enum value has exactly one bit set). Otherwise we would need `self as u8 & stored_bits == self as u8` or something like that.

Reply:
Yes, and I don't see any signs of a AND-style permission in the future that would require your proposed comparison. So should be good.

Reply:
I'll add documentation to clearly state that intention.